### ISSUE TITLE
public bucket is not accessible

### ISSUE THREAD

**Issue by regevbr** [binxio/aws-cloudwatch-log-minder#2]
Hi,

Thanks for this great tool!

I have a few suggestions and issues:

1. It will be easier to use a single line to deploy with cloud formation:
```bash
aws cloudformation deploy --stack-name=aws-cloudwatch-log-minder --template-file=./cloudformation/aws-cloudwatch-log-minder.yaml --capabilities CAPABILITY_IAM --parameter-overrides LambdaS3Bucket=<my bucket> CFNCustomProviderZipFileName=<zipfile location>
```
2. Add cloudformation parameter to change the default retention period
3. List all the required permissions for the cloudformation stack creation
4. The public s3 bucket that contains the zipped version is not publicly accessible - I had to manually build it (using make) and deploy it to my own bucket
5. When trying to build manually, the make command failed because you pass the `encoding` param in `setup.py` to the open method (line 10), and apparently it is not supported. Removing this param solved the issue for me.

side note - this is the first time I dealt with python code :-)

@mvanholsteijn any thoughts?

**Comment by mvanholsteijn** [comment]
@regevbr, you title says that the bucket is not accessible. What region are you referring to?

```
$ aws s3 ls s3://binxio-public-eu-central-1/lambdas/ | grep minder-0.2.3
2019-09-13 17:05:31      13188 aws-cloudwatch-log-minder-0.2.3.zip
```
In addition, you have reported 5 issues in 1 :-)

For #1 I do not see now deploy is any better dan create-stack
For #2 I would suggest that you put in as PR.
For #3 it is unclear what you mean. The readme clearly states you have to run `--capabilities CAPABILITY_IAM`
For #4 please state which bucket in which region you find is not accessible.
For #5 I do not have this error. what platform are you running on? Which version of python are you using? What is the exact error?

**Comment by regevbr** [comment]
Hi,

Sorry for the mess, it was 2am on my side, and all those issues piled up while I was trying (for 1 hour) to get it working.

1. It is better for a number of reasons:
 - It it consolidates your 2 commands (create-stack and wait) into a single one
 - It supports updating the stack after it was already deployed incase you make changes to the parameters
 - It is easier to pass the custom parameters to it
2. Sure thing, I will work on it.
3. I was talking about the permissions that the user calling the command needs - in the off chance you run the cloudformation command with a non admin user, you need to have a lot of permissions, such as creating lambda functions etc... I had to work my way very slowly to get all those permissions right before I could make the stack get created. Sadly I don't have the list of those permissions as I allowed a * permissions on the required services.
4. I'm talking about eu-centeral-1. I didn't say it doesn't exist, just that it is not publicly accessible. Trying to deploy the stack without stating the custom parameters, it fails because the bucket (your "public" one) is not publicly accessible. (go into S3 console, get the url path of the file and try to open it in incognito mode and it will fail)
5. Sorry, my pipenev got mixed up and actually ran with python 2.7 instead of 3.6, please ignore this issue.

**Comment by mvanholsteijn** [comment]
Ok. I fixed `#1` in commit 12396bd

`#3` is one I never do. It would be too much hassle. 

`#4` both the bucket and the object are copied with the acl public-read. The curl works, which is as public as it gets.

```
curl https://binxio-public-eu-central-1.s3.eu-central-1.amazonaws.com/lambdas/aws-cloudwatch-log-minder-0.2.3.zip
```

**Comment by regevbr** [comment]
Thanks for all the fixes! About the bucket it is very strange as I tried that url yesterday without luck, I will update here if it still imposes an issue for me
